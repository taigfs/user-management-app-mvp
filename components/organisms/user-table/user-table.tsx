import { View, Text } from "react-native";

import { DeleteUserDialog } from "../delete-user-dialog";
import { EditUserDialog } from "../edit-user-dialog";

import { TableField } from "@/types/table-field";
import { User } from "@/types/user";

interface UserTableProps {
  data: User[];
  fields: TableField[];
}

export const UserTable = ({ data, fields }: UserTableProps) => {
  return (
    <View className="w-full table-auto">
      <View>
        <View className="bg-muted text-muted-foreground flex-row">
          {fields.map((field) => (
            <Text
              key={field.key}
              className={`px-2 py-2 text-left text-white flex-${field.flex}`}
            >
              {field.label}
            </Text>
          ))}
          <Text className="px-2 py-2 flex-2 text-white">Actions</Text>
        </View>
      </View>
      <View>
        {data.map((user) => (
          <View key={user.id} className="border-b flex-row">
            <View className="px-2 py-2 flex-3">{user.name}</View>
            <View className="px-2 py-2 flex-4">{user.email}</View>
            <View className="px-2 py-2 flex-4">{user.phone}</View>
            <View className="px-2 py-2 flex-2 flex-row">
              <EditUserDialog />
              <DeleteUserDialog />
            </View>
          </View>
        ))}
      </View>
    </View>
  );
};
