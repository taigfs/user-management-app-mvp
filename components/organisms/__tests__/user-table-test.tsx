import React from "react";
import renderer from "react-test-renderer";

import { UserTable } from "../user-table/user-table";

it(`renders correctly`, () => {
  const tree = renderer.create(<UserTable data={[]} fields={[]} />).toJSON();

  expect(tree).toMatchSnapshot();
});
