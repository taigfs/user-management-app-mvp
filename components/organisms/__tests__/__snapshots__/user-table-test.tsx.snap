// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  className="w-full table-auto"
>
  <View>
    <View
      className="bg-muted text-muted-foreground flex-row"
    >
      <Text
        className="px-2 py-2 flex-2 text-white"
      >
        Actions
      </Text>
    </View>
  </View>
  <View />
</View>
`;
